### PR TITLE
dev-db/mongodb: bump and fixes

### PR DESCRIPTION
**Pull request by hydrapolic**
This fixes building with glibc 2.25-r4 for both 3.2 and 3.4 (https://bugs.gentoo.org/show_bug.cgi?id=575232) and backporting the patch for pcre in 3.2.

https://bugs.gentoo.org/show_bug.cgi?id=580768
https://bugs.gentoo.org/show_bug.cgi?id=625148
https://bugs.gentoo.org/show_bug.cgi?id=626342
https://bugs.gentoo.org/show_bug.cgi?id=630566

### CODE REVIEW

File: app-admin/mongo-tools/mongo-tools-3.4.7.ebuild [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (monsieurp) [inline app-admin/mongo-tools/mongo-tools-3.4.7.ebuild]: EAPI 6 pls.

File: app-admin/mongo-tools/mongo-tools-3.4.7.ebuild [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+S=${WORKDIR}/${MY_P}
+
+src_prepare() {
+	sed -e 's|go build .*|go build -o "bin/$i" -tags "$tags" "$i/main/$i.go"|g' -i build.sh || die
+
+	# ensure we use bash wrt #582906
+	sed -e 's@/bin/sh@/bin/bash@g' -i build.sh || die
+
+	epatch "${FILESDIR}/${PN}-3.2.10-pie.patch"
```
Comment (monsieurp) [inline app-admin/mongo-tools/mongo-tools-3.4.7.ebuild]: call `default` and add this patch to the `PATCHES` array.

File: app-admin/mongo-tools/mongo-tools-3.4.7.ebuild [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+src_compile() {
+	local myconf
```
Comment (monsieurp) [inline app-admin/mongo-tools/mongo-tools-3.4.7.ebuild]: please use an array instead. it is way easier to append elements to an array than to redefine a string.

File: dev-db/mongodb/mongodb-3.2.16-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		filter-flags '-m*'
+		filter-flags '-O?'
+	fi
+	escons "${scons_opts[@]}" core tools
+}
+
+src_install() {
+	escons "${scons_opts[@]}" --nostrip install --prefix="${ED}"/usr
+
+	for x in /var/{lib,log}/${PN}; do
```
Comment (monsieurp) [inline dev-db/mongodb/mongodb-3.2.16-r1.ebuild]: `local x`

File: dev-db/mongodb/mongodb-3.2.16-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+pkg_preinst() {
+	# wrt bug #461466
+	if [[ "$(get_libdir)" == "lib64" ]]; then
+		rmdir "${ED}"/usr/lib/ &>/dev/null
+	fi
+}
+
+src_test() {
+	# this one test fails
+	rm jstests/core/repl_write_threads_start_param.js
```
Comment (monsieurp) [inline dev-db/mongodb/mongodb-3.2.16-r1.ebuild]: `|| die`

File: dev-db/mongodb/mongodb-3.2.16-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,195 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (monsieurp) [inline dev-db/mongodb/mongodb-3.2.16-r1.ebuild]: EAPI 6 pls.

File: dev-db/mongodb/mongodb-3.2.16-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+pkg_setup() {
+	enewgroup mongodb
+	enewuser mongodb -1 -1 /var/lib/${PN} mongodb
+
+	# Maintainer notes
+	#
+	# --use-system-tcmalloc is strongly NOT recommended:
+	# https://www.mongodb.org/about/contributors/tutorial/build-mongodb-from-source/
+
+	scons_opts=(
```
Comment (monsieurp) [inline dev-db/mongodb/mongodb-3.2.16-r1.ebuild]: This should be defined as `local`.

File: dev-db/mongodb/mongodb-3.2.16-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+	if use kerberos; then
+		scons_opts+=( --use-sasl-client )
+	fi
+
+	if use ssl; then
+		scons_opts+=( --ssl )
+	fi
+}
+
+src_prepare() {
```
Comment (monsieurp) [inline dev-db/mongodb/mongodb-3.2.16-r1.ebuild]: As usual, `default` + patches in `PATCHES`.

File: dev-db/mongodb/mongodb-3.2.16-r1.ebuild (hunk 7 above)
Comment (hydrapolic) [reply dev-db/mongodb/mongodb-3.2.16-r1.ebuild]: It's used later on so probably not.

**Comment by hydrapolic** [comment]
@monsieurp, thanks for the review, updated.

**Comment by dman777** [comment]
Need this patch also. If someone can please approve and merge.

**Comment by hydrapolic** [comment]
Thanks @monsieurp, hope this works out well :)

**Pull request closed without merging**